cmd/kubectl-gadget/deploy.go

```diff
@@ -453,6 +467,16 @@ func runDeploy(cmd *cobra.Command, args []string) error {
 				}
 			}
 
+			// When using bcc flavor, we may need extra capabilities.
+			// Particularly, we should add the SYS_MODULE one, as this is needed by
+			// bcc gadgets to load kheaders.ko in case it is not built-in.
+			// NOTE Talos forbids this capability:
+			// https://support.siderolabs.com/hc/en-us/articles/18362930553876-Process-Capabilities
+			// So, for Talos, only default image works.
+			if strings.HasSuffix(image, "bcc") {
```

Comment:
What is the reasoning to base this on the image name and not in the `bcc` flag? 

For instance, when trying this locally, I noticed it misbehaved in my setup because the image tag created by this branch is `192.168.1.150:5000/gadget:francis-deploy-bcc` (the -bcc is in the branch name, it's not the bcc image), it was wrongly adding the sys_module capability when not needed.